### PR TITLE
Repay debt without collateral withdrawal

### LINKED ISSUE #481: Proxy Repay Debt without Withdrawing
Repay debt is prevented without withdrawing some amount of collateral

### PR DESCRIPTION
**Pull request by MrDeadCe11**
closes #481

Deployed a new BasicActions contract to use taxSingle during repayDebt and to use the correct allowance modifications when modifying ERC20 allowance.

also added my name to the contributors list in the package.json.  :-)

sorry about the commits that made it in from the last branch I worked on.  but those have been merged to dev already.

### CODE REVIEW

File: src/contracts/proxies/actions/BasicActions.sol [hunk 1]
```diff
@@ -320,7 +320,7 @@ contract BasicActions is CommonActions, IBasicActions {
     _taxSingle(_manager, _safeId);
 
     ISAFEEngine.SAFE memory _safeData = ISAFEEngine(_safeEngine).safes(_safeInfo.collateralType, _safeInfo.safeHandler);
-    _taxSingle(_manager, _safeId);
+
```
Comment (pi0neerpat) [inline src/contracts/proxies/actions/BasicActions.sol:323]: I understood your message to mean we needed to add this back, not remove it? Is intended to be kept?

File: src/contracts/proxies/actions/CommonActions.sol [hunk 2]
```diff
@@ -56,8 +56,15 @@ abstract contract CommonActions is ICommonActions {
...
-    // Approves adapter to take the COIN amount
-    SafeERC20.safeApprove(IERC20(_systemCoin), _coinJoin, _wad);
+    //check if coinJoin has an approval amount
+    uint256 allowance = IERC20(_systemCoin).allowance(address(this), _coinJoin);
+    if (allowance == 0) {
+      // Approves adapter to take the COIN amount
+      SafeERC20.safeApprove(IERC20(_systemCoin), _coinJoin, _wad);
+    } else {
+      // if approval amount is a non-zero number, increase allowance.
+      SafeERC20.safeIncreaseAllowance(IERC20(_systemCoin), _coinJoin, _wad);
```
Comment (pi0neerpat) [inline src/contracts/proxies/actions/CommonActions.sol:66]: I can't think of any security implications here. If the tx fails, then the allowance isn't increased, therefore there is no risk of having allowance higher than necessary.

**Review by pi0neerpat — APPROVED** [review]
If you deployed new Common and Basic Actions, then we need to update the SDK to use these. If I understood correctly, we can update these two contract addresses, and the issue can be resolve in the current deployment of the protocol, without having to re-deploy everything.

- Provide the addresses for the new action contracts on Arb Sepolia

File: src/contracts/proxies/actions/CommonActions.sol (hunk 2 above)
Comment (MrDeadCe11) [reply src/contracts/proxies/actions/CommonActions.sol:66]: that's up to you.  we could make it to check and modify the allowance to only be enough to allow the TX in question if that's the more desired solution.  e.g. 
```if(allowance < desiredAllowance) {
_wad = desiredAllowance - allowance;
SafeERC20.safeIncreasAllowance(IERC20(_systemCoin), _coinJoin, _wad);
} ```

**Comment by MrDeadCe11** [comment]
> If you deployed new Common and Basic Actions, then we need to update the SDK to use these. If I understood correctly, we can update these two contract addresses, and the issue can be resolve in the current deployment of the protocol, without having to re-deploy everything.
> 
> * Provide the addresses for the new action contracts on Arb Sepolia

yes, this is accurate.  I updated the addresses in the SeploiaContracts.s.sol with the working BasicActions.sol contract.
the CommonActions.sol is an abstract so it only gets re-deployed as part of the BasicActions.sol deployment.

**Pull request merged**